src/core/markovProcessGraph.js

```diff
@@ -0,0 +1,579 @@
+        epochStart: epochTimestampMs,
+      });
+    };
+
+    // Add graph edges. First, split by direction.
+    type _UnidirectionalGraphEdge = {|
+      +address: EdgeAddressT,
+      +reversed: boolean,
+      +src: NodeAddressT,
+      +dst: NodeAddressT,
+      +timestamp: TimestampMs,
+      +weight: number,
+    |};
+    const unidirectionalGraphEdges = function* (): Iterator<_UnidirectionalGraphEdge> {
+      const ewe = edgeWeightEvaluator(wg.weights);
+      for (const edge of (function* () {
+        for (const edge of wg.graph.edges({showDangling: false})) {
+          const weight = ewe(edge.address);
+          yield {
+            address: edge.address,
+            reversed: false,
+            src: edge.src,
+            dst: edge.dst,
+            timestamp: edge.timestampMs,
+            weight: weight.forwards,
+          };
+          yield {
+            address: edge.address,
+            reversed: true,
+            src: edge.dst,
+            dst: edge.src,
+            timestamp: edge.timestampMs,
+            weight: weight.backwards,
+          };
+        }
+      })()) {
+        if (edge.weight > 0) {
+          yield edge;
+        }
+      }
+    };
+
+    const srcNodes: Map<
+      NodeAddressT /* domain: (nodes with out-edges) U (epoch nodes) */,
+      {totalOutWeight: number, outEdges: _UnidirectionalGraphEdge[]}
+    > = new Map();
+    for (const graphEdge of unidirectionalGraphEdges()) {
+      const src = rewriteEpochNode(graphEdge.src, graphEdge.timestamp);
+      let datum = srcNodes.get(src);
+      if (datum == null) {
+        datum = {totalOutWeight: 0, outEdges: []};
+        srcNodes.set(src, datum);
+      }
+      datum.totalOutWeight += graphEdge.weight;
+      datum.outEdges.push(graphEdge);
+    }
+    for (const [src, {totalOutWeight, outEdges}] of srcNodes) {
+      const totalOutPr = NodeAddress.hasPrefix(src, EPOCH_PREFIX)
+        ? epochTransitionRemainder
+        : 1 - seed.alpha;
+      for (const outEdge of outEdges) {
+        const pr = (outEdge.weight / totalOutWeight) * totalOutPr;
+        addEdge({
+          address: outEdge.address,
+          reversed: outEdge.reversed,
+          src: rewriteEpochNode(outEdge.src, outEdge.timestamp),
+          dst: rewriteEpochNode(outEdge.dst, outEdge.timestamp),
+          transitionProbability: pr,
+        });
+      }
+    }
+
+    return new MarkovProcessGraph(_nodes, _edges, _scoringAddresses);
+  }
+
+  scoringAddresses(): Set<NodeAddressT> {
+    return new Set(this._scoringAddresses);
+  }
+
+  node(address: NodeAddressT): MarkovNode | null {
+    NodeAddress.assertValid(address);
+    return this._nodes.get(address) || null;
+  }
+
+  *nodes(options?: {|+prefix: NodeAddressT|}): Iterator<MarkovNode> {
+    const prefix = options ? options.prefix : NodeAddress.empty;
+    for (const node of this._nodes.values()) {
+      if (NodeAddress.hasPrefix(node.address, prefix)) {
+        yield node;
+      }
+    }
+  }
```

Comment:
Lately I've been thinking we should just return arrays rather than iterators--i cant recall ever benefiting from using the iterate and it's a bit more cumbersome. 

no change requested.